### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by jaamanu**
In this PR we 
Add unit tests for all the methods in all the classes:
Book
Person
Student
Teacher
Classroom
Rental
Fix linter errors

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -1,118 +1,140 @@
...
+      [1] List all books
+      [2] List all people
+      [3] Add a person
+      [4] Add a book
+      [5] Add a rental
+      [6] List all rentals for a given person id
+      [7] Exit
+      Select an option by typing its number: "
+  end
+
```
Comment (Davidosky007) [inline app.rb:25]: * Kindly add unit tests for all the methods in all the classes:

> Book
> Person
> Student
> Teacher
> Classroom
> Rental
> all classes that you created while implementing the Decorator design pattern
* As I can not find any [Check this lesson for a refresher on unit testing](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/rspec_unit_test.md)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @jaamanu @KareemWilson 👋, 
Almost-gif-simpson

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* well written PR description 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (rloterh) [reply app.rb:25]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Kareem & James,  

Your project is complete! 🏁 Nice job! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
- Correctly added unit test for various classes~

## Optional suggestions 🧰
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
- Kindly ensure that your `Gemfile` is correctly named instead of `Gemefile`.
- By conversion your test directory should be labeled `spec` instead of `rspec`.
- Kindly ensure to add the _Future Features_ section to your ReadMe, which should include 1-3 potential features to be added.
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**